### PR TITLE
NoRecentEmoji implementation. Fixes #477

### LINKED ISSUE #477: Disable RecentEmoji
Is there any way to disable the RecentEmoji tab? (version 0.7.0)
I want to dismiss the Recent Emojis

### PR DESCRIPTION
**Pull request by vanniktech**
Fixes #477

### CODE REVIEW

**Comment by vanniktech** [comment]
@rubengees @mario I'd like to cut a new release tomorrow with this, so if you want to review, please do :)

File: emoji/src/main/java/com/vanniktech/emoji/EmojiPagerAdapter.java [hunk 1]
```diff
@@ -33,44 +34,54 @@
...
     this.recentEmoji = recentEmoji;
     this.variantManager = variantManager;
-    this.recentEmojiGridView = null;
+  }
+
+  boolean hasRecentEmoji() {
+    return !(recentEmoji instanceof NoRecentEmoji);
+  }
+
+  int recentCount() {
```
Comment (rubengees) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiPagerAdapter.java:53]: This name confused me, I thought this means how many recent emojis the user has saved from using the component previously. Maybe inline this in the `EmojiView` constructor or find a different name?

**Review by rubengees — APPROVED** [review]
Look's good to me :+1:

**Comment by mario** [comment]
Can Review tonight if That’s ok

**Comment by vanniktech** [comment]
@mario sure!

File: emoji/src/main/java/com/vanniktech/emoji/EmojiPagerAdapter.java (hunk 1 above)
Comment (mario) [reply emoji/src/main/java/com/vanniktech/emoji/EmojiPagerAdapter.java:53]: Have to agree here, do we need this method?

File: emoji/src/main/java/com/vanniktech/emoji/EmojiPagerAdapter.java [hunk 2]
```diff
@@ -33,44 +34,54 @@
...
+    return hasRecentEmoji() ? 1 : 0;
   }
 
   @Override public int getCount() {
-    return EmojiManager.getInstance().getCategories().length + 1;
+    return EmojiManager.getInstance().getCategories().length + recentCount();
```
Comment (mario) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiPagerAdapter.java:58]: Why not simply +1 if hasRecentEmoji without using recentCount?

File: emoji/src/main/java/com/vanniktech/emoji/EmojiPagerAdapter.java (hunk 1 above)
Comment (vanniktech) [reply emoji/src/main/java/com/vanniktech/emoji/EmojiPagerAdapter.java:53]: Yeah it's being used multiple times (also for the top bar, when we don't have that recent emoji tab)

Does anyone have a better name?

**Comment by vanniktech** [comment]
I'll merge this. If you have another name suggestion I can change it afterwards :)

**Pull request merged**